appgate/files.py

```diff
@@ -71,19 +76,32 @@ def url_file_path(
     4. else if the entity has an id, use it to compute the path url `id/fieldName`
     5. else raise an exception
     """
-    if (v := value.get(field_name)) is not None:
+    v = value.get(field_name)
+    if v is not None and isinstance(v, str):
         return normalize_url_file_path(v)
     for field in target_fields:
         if v := value.get(field):
             return v
     if value.get("name"):
         return f"{normalize_url_file_path(value['name'])}/{field_name}"
-    elif value.get("id"):
+    if value.get("id"):
         return f"{value['id']}/{field_name}"
-    else:
-        raise AppgateFileException(
-            f"Unable to generate url to get fetch file for field {field_name} for entity {entity_name}"
-        )
+
+    # TODO: Problematic entities. Below are hardcoded workarounds.
+    if entity_name == "Appliance_Ntp_Servers":
```

Comment:
Good enough for now but we should implement a generic `x-field` to catch this kind of problems